### ISSUE TITLE
Compile error with PHP 8.1

### ISSUE THREAD

**Issue by DoM1niC** [zeroc-ice/ice#1317]
**Version(please complete the following information):**
 - Ice Version 3.7.4
 - OS: Linux (Debian Bookworm)
 - Distribution: apt

**Describe the bug**
Compiling error

**Expected behavior**
```
/usr/include/php/20210902/Zend/zend_compile.h:820:15: error: redundant redeclaration of ‘void zend_destroy_file_handle(zend_file_handle*)’ in same scope [-Werror=redundant-decls]
  820 | ZEND_API void zend_destroy_file_handle(zend_file_handle *file_handle);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/20210902/Zend/zend.h:38,
                 from /usr/include/php/20210902/main/php.h:32,
                 from src/php/Config.h:67,
                 from src/php/Types.h:8,
                 from src/php/Types.cpp:5:
/usr/include/php/20210902/Zend/zend_stream.h:71:15: note: previous declaration of ‘void zend_destroy_file_handle(zend_file_handle*)’
   71 | ZEND_API void zend_destroy_file_handle(zend_file_handle *file_handle);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/20210902/Zend/zend.h:38,
                 from /usr/include/php/20210902/main/php.h:32,
                 from src/php/Config.h:67,
                 from src/php/Proxy.h:8,
                 from src/php/Proxy.cpp:5:
/usr/include/php/20210902/Zend/zend_stream.h:71:15: note: previous declaration of ‘void zend_destroy_file_handle(zend_file_handle*)’
   71 | ZEND_API void zend_destroy_file_handle(zend_file_handle *file_handle);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/20210902/Zend/zend.h:38,
                 from /usr/include/php/20210902/main/php.h:32,
                 from src/php/Config.h:67,
                 from src/php/Operation.h:8,
                 from src/php/Operation.cpp:5:
/usr/include/php/20210902/Zend/zend_stream.h:71:15: note: previous declaration of ‘void zend_destroy_file_handle(zend_file_handle*)’
   71 | ZEND_API void zend_destroy_file_handle(zend_file_handle *file_handle);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/php/20210902/Zend/zend.h:38,
                 from /usr/include/php/20210902/main/php.h:32,
                 from src/php/Config.h:67,
                 from src/php/Connection.h:8,
                 from src/php/Connection.cpp:5:
/usr/include/php/20210902/Zend/zend_stream.h:71:15: note: previous declaration of ‘void zend_destroy_file_handle(zend_file_handle*)’
   71 | ZEND_API void zend_destroy_file_handle(zend_file_handle *file_handle);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Init.o] Error 1
make[1]: *** Waiting for unfinished jobs....
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Logger.o] Error 1
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Endpoint.o] Error 1
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Properties.o] Error 1
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Connection.o] Error 1
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Util.o] Error 1
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Operation.o] Error 1
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Proxy.o] Error 1
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Communicator.o] Error 1
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:39: src/php/build/x86_64-linux-gnu/shared/pic/Types.o] Error 1
make[1]: Leaving directory '/mnt/e/Git/ice/php'
make: *** [Makefile:20: all] Error 1
```

**Comment by pepone** [comment]
Hi @DoM1niC 

I think this was fixed in 3.7.5 see https://github.com/zeroc-ice/ice/issues/1172. I just tested PHP 8.1 on macOS and seems to build fine.

**Comment by bryceharrington** [comment]
I'm reproducing this same error on Ubuntu using 3.7.6-2, with the openssl3.0 and gpp 11.2 fixes applied.
I notice the log mentions -Wredundant-decls, which appears to be set for Make.rules.[Linux|Darwin] but not other architectures, so that might be worth checking?
I'm going to test if it builds with that option dropped from cppflags.

**Comment by pepone** [comment]
@bryceharrington what Ubuntu version are you testing on?

**Comment by bryceharrington** [comment]
Hi pepone, this is on ubuntu 22.04 (jammy-proposed), with php8.1 and openssl3.0.

**Comment by pepone** [comment]
Fixed in f542ad5eb07d9f129d7160d9e46ab07bda2afa7a, uploaded 3.7.6-4 package with patch
